files/image_config/procdockerstats/procdockerstats

```diff
@@ -0,0 +1,155 @@
+                    dict1 =  dict(zip(keys, values1))
+                    dict_list.append(dict1)
+                return dict_list                                        
+                                     
+    def update_process_docker_commands(self):
+        """
+        Convenience wrapper which retrieves process and docker CPU/memory utilization data
+        """
+        process_docker_cmds = []
+        process_docker_cmds.append("docker stats --no-stream -a")
+        """ process_docker_cmds.append("ps aux")"""
+       
+        log_info("Issuing the following commands:")
+        for cmd in  process_docker_cmds:
+            log_info("  " + cmd)
+
+        dockerdata = self.run_commands(process_docker_cmds)
+        value = ""
+
+        for index in range(len(dockerdata)):
+            row = dockerdata[index]            
+            cid = row.get('CONTAINER ID')
+            if cid != "":               
+                value = 'Docker_Stats|' + str(cid)
+                name = row.get('NAME')
+                self.update_state_db(value, 'NAME', name)
+
+                cpu = row.get('CPU %')
+                self.update_state_db(value, 'CPU%', str(cpu))
+
+                splitcol = row.get('MEM USAGE / LIMIT')
+                memuse = re.split(" / ", str(splitcol))
+                self.update_state_db(value, 'MEM', str(memuse[0]))
+                self.update_state_db(value, 'MEM_LIMIT', str(memuse[1]))
+
+                mem = row.get('MEM %')
+                self.update_state_db(value, 'MEM %', str(mem))
+
+                splitcol = row.get('NET I/O')
+                netio = re.split(" / ", str(splitcol))
+                self.update_state_db(value, 'NET_IN', str(netio[0]))
+                self.update_state_db(value, 'NET_OUT', str(netio[1]))
+
+                splitcol = row.get('BLOCK I/O')
+                blocio = re.split(" / ", str(splitcol))
+                self.update_state_db(value, 'BLOCK_IN', str(blocio[0]))
+                self.update_state_db(value, 'BLOCK_OUT', str(blocio[1]))
+
+                pids = row.get('PIDS')
+                self.update_state_db(value, 'PIDS', pids)            
+
+    def update_state_db(self, key1, key2, value2):
+        self.state_db.set('STATE_DB', key1, key2, value2)
+
+    def run(self):
+        self.update_process_docker_commands()
+        datetimeobj = datetime.now()
+        # Adding key to store latest update time.
+        self.update_state_db('Docker_Stats|LastUpdateTime', 'lastupdate', datetimeobj)
+
+
+
+# main start
+def main():
+    log_info("process-docker stats aemon starting up..")
+
+    if not os.getuid() == 0:
+        log_error("Must be root to run process-docker daemon")
+        print "Error: Must be root to run process-docker daemon"
+        sys.exit(1)
+
+    pd = ProcDockerStats()
+    pd.initialize()
+    # Data need to be updated every 2 mins. hence adding delay of 120 seconds
+    #time.sleep(120)
```

Comment:
I assume this will be uncommented later.

Reply:
yes my bad.